servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java

```diff
@@ -10098,6 +10137,144 @@ public void deleteAssertionCondition(ResourceContext ctx, String domainName, Str
+        // Otherwise ignore it and use the principal as dependent service
+
+        Principal principal = ((RsrcCtxWrapper) ctx).principal();
+        boolean isSysAdminUser = isSysAdminUser(principal);
+        if (isSysAdminUser) {
+            if (StringUtil.isEmpty(service)) {
+                throw ZMSUtils.requestError("service is mandatory", caller);
+            }
+        } else {
+            service = principal.getFullName();
+        }
+
+        validateRequest(ctx.request(), caller);
+
+        validate(domainName, TYPE_DOMAIN_NAME, caller);
+        validate(service, TYPE_SERVICE_NAME, caller);
+
+        // for consistent handling of all requests, we're going to convert
+        // all incoming object values into lower case (e.g. domain, role,
+        // policy, service, etc name)
+
+        domainName = domainName.toLowerCase();
+        setRequestDomain(ctx, domainName);
+        service = service.toLowerCase();
+
+        // verify that request is properly authenticated.
+        // system administrator can delete any dependency, even for services no longer listed as a service providers
+
+        if (!isSysAdminUser && !serviceProviderManager.isServiceProvider(service)) {
+            throw ZMSUtils.unauthorizedError("Principal is not a System Administrator or Authorized Service Provider", caller);
+        }
+
+        // Verify the domain exist
+
+        AthenzDomain domain = getAthenzDomain(domainName, false);
+        if (domain == null || domain.getDomain() == null) {
+            throw ZMSUtils.notFoundError("Invalid domain: " + domainName, caller);
+        }
+
+        dbService.deleteDomainDependency(ctx, domainName, service, auditRef, caller);
+    }
+
+    @Override
+    public ServiceIdentityList getDependentServiceList(ResourceContext ctx, String domainName) {
+        final String caller = ctx.getApiName();
+        logPrincipal(ctx);
+
+        validateRequest(ctx.request(), caller);
+        validate(domainName, TYPE_DOMAIN_NAME, caller);
+
+        // for consistent handling of all requests, we're going to convert
+        // all incoming object values into lower case (e.g. domain, role,
+        // policy, service, etc name)
+
+        domainName = domainName.toLowerCase();
+        setRequestDomain(ctx, domainName);
+
+        return dbService.listServiceDependencies(domainName);
+    }
+
+    @Override
+    public DomainList getDependentDomainList(ResourceContext ctx, String service) {
+        final String caller = ctx.getApiName();
+        logPrincipal(ctx);
+
+        validateRequest(ctx.request(), caller);
+        validate(service, TYPE_SERVICE_NAME, caller);
+
+        // for consistent handling of all requests, we're going to convert
+        // all incoming object values into lower case (e.g. domain, role,
+        // policy, service, etc name)
+
+        service = service.toLowerCase();
+
+        return dbService.listDomainDependencies(service);
```

Comment:
before calling this, let's add an extra validation to make sure the service is a registered service provider. otherwise anyone can just call an api with some random service name

Reply:
Similar to my previous comment here https://github.com/AthenZ/athenz/pull/1758#discussion_r798418241, it is possible for a service provider to be removed from the list of registered service providers (our management role) in which case it will still have dependencies and they will not appear if we do this check.

Reply:
Fixed